Source/Scene/ShadowMap.js

```diff
@@ -64,33 +83,70 @@ define([
         OrthographicFrustum,
         Pass,
         PerInstanceColorAppearance,
+        PerspectiveFrustum,
         Primitive) {
     'use strict';
 
-    // TODO : doc
     /**
      * @private
      */
-    function ShadowMap(context, camera) {
+    function ShadowMap(context, lightCamera) {
         this.enabled = false;
 
+        // Uniforms
+        this._shadowMapMatrix = new Matrix4();
+        this._shadowMapTexture = undefined;
+
+        this._framebuffer = undefined;
+        this._shadowMapSize = 1024;
+
+        this._lightCamera = lightCamera;
+        this._shadowMapCamera = new ShadowMapCamera();
+        this._sceneCamera = undefined;
+        this._farPlane = 100.0; // Limit the far plane of the scene camera
+        this._fitToScene = true;
+
+        var numberOfCascades = 4;
```

Comment:
Yes this should be customizable as well as the cascade split distances. Right now the shadow map is somewhat rigid, either supporting one cascade or four cascades, so this value is more of a reference than anything. Supporting two or three is not bad, just a matter of changing the texture atlas size / texture offsets. Above four cascades is less supported, mainly because I use vec4's in the shader code.